### PR TITLE
[Woo POS] [Variable Products] Show POS with `fullScreenCover` with its own navigation stack

### PR DESCRIPTION
**Pull request by jaclync**
First iteration of #14683 

## Why

From @joshheald's technical post pdfdoF-61P-p2:

> Using NavigationStack requires that we change how the POS is presented. Currently it is pushed onto the HubMenu‘s navigation stack, and we can’t nest them.

As a simple solution, we could present the POS view using `fullScreenCover` from the HubMenu view with the native present/dismiss animation from the bottom. Unfortunately, I haven't found a way to customize the animation for `fullScreenCover`. If we want to customize the transition to/from POS later, we could consider using a ZStack like in this exploration https://github.com/woocommerce/woocommerce-ios/commit/b29fc32043b34bf5a96696b7fb75a027d27b2636. I tested this approach with the experiment branch in https://github.com/woocommerce/woocommerce-ios/commit/f9656cbc6001b51ed3008c46149458e7855e5106 and the navigation between the products list and variations list works as expected under the POS' navigation stack.

With the changes in this PR, it does change how the POS is presented from the push navigation (horizontal transition) to the presentation navigation (vertical transition). @joshheald Please let me know if I missed any key reasons why a `UIWindow` in your WIP branch https://github.com/woocommerce/woocommerce-ios/pull/14674 is required (the window has no transition animation in the branch yet).

## How

The changes include adding a new full-screen cover for the POS entry point, removing the old POS navigation destination, and updating the view model to manage the POS state.

Key changes include:

### Improvements to POS handling:

* [`WooCommerce/Classes/ViewRelated/Hub Menu/HubMenu.swift`](diffhunk://#diff-3005ff4e49c7b74962724c77a4a606b4398af51e4726bda1ae390220cf67fc8cR34-R49): Added a full-screen cover to present the POS entry point when `showsPOS` is true. This includes necessary checks and error handling for creating the card payment service and order service.
* [`WooCommerce/Classes/ViewRelated/Hub Menu/HubMenu.swift`](diffhunk://#diff-3005ff4e49c7b74962724c77a4a606b4398af51e4726bda1ae390220cf67fc8cR65-R66): Updated the action for the POS entry point to set `showsPOS` to true, triggering the new full-screen cover.
* [`WooCommerce/Classes/ViewRelated/Hub Menu/HubMenu.swift`](diffhunk://#diff-3005ff4e49c7b74962724c77a4a606b4398af51e4726bda1ae390220cf67fc8cL157-L171): Removed the old case for handling the POS entry point within the existing navigation structure.

### View model updates:

* [`WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift`](diffhunk://#diff-a7353a3c9b54e18ce66e192000f034059a624c6824a359575150dc1788697d04L31): Removed the `pointOfSales` case from the `HubMenuNavigationDestination` enum as it is no longer needed.
* [`WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift`](diffhunk://#diff-a7353a3c9b54e18ce66e192000f034059a624c6824a359575150dc1788697d04R85-R87): Added a new `showsPOS` property to manage the state of the POS feature and updated the POS menu item to have no navigation destination. [[1]](diffhunk://#diff-a7353a3c9b54e18ce66e192000f034059a624c6824a359575150dc1788697d04R85-R87) [[2]](diffhunk://#diff-a7353a3c9b54e18ce66e192000f034059a624c6824a359575150dc1788697d04L663-R666)

## Steps to reproduce

- Switch to a store eligible for POS
- Go to Menu > Point of Sale mode --> the POS view should be presented from bottom to top, this is a change from the push navigation in trunk
- Feel free to perform any actions in POS
- Tap eliipsis menu in the bottom bar > Exit POS > Exit --> the POS view should be dismissed from top to bottom, this is a change from the pop navigation in trunk

## Screenshots

Before:

https://github.com/user-attachments/assets/c2b3dc6a-5feb-43a8-b1b4-c0c96e9c6730

After:

https://github.com/user-attachments/assets/3be110f0-5087-48f0-8e32-190a75c0f41b

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Hub Menu/HubMenu.swift [hunk 1]
```diff
@@ -31,6 +31,22 @@ struct HubMenu: View {
...
                     viewModel.setupMenuElements()
                 }
+                .fullScreenCover(isPresented: $viewModel.showsPOS) {
+                    if let cardPresentPaymentService = viewModel.cardPresentPaymentService,
+                       let orderService = POSOrderService(siteID: viewModel.siteID,
+                                                          credentials: viewModel.credentials) {
+                        PointOfSaleEntryPointView(
+                            itemsController: PointOfSaleItemsController(itemProvider: viewModel.posItemProvider),
+                            onPointOfSaleModeActiveStateChange: { isEnabled in
+                                viewModel.updateDefaultConfigurationForPointOfSale(isEnabled)
```
Comment (staskus) [inline WooCommerce/Classes/ViewRelated/Hub Menu/HubMenu.swift:41]: `updateDefaultConfigurationForPointOfSale` calls `updateTabBarVisibility` which may not be needed as we present a full screen cover

**Comment by jaclync** [comment]
I'm double-checking if there was a prior decision on the transition from the Menu tab to POS mode in p1734331128921509-slack-C070SJRA8DP. There's likely none because Android has a different crossfade-like transition from iOS.

**Review by staskus — APPROVED** [review]
It works well, tried presenting other modals, support, go to order management from simple product notice. I cannot however make a comment about the integration with variable products and if there are no technical blockers. I'll leave @joshheald to comment about this.

**Review by joshheald — APPROVED** [review]
Joe didn't love the slide-up transition... I think he wanted it to feel more like a mode switch. However, in the end we just went with the quickest thing, so I don't think the existing transition is thoroughly considered.

If we need to customise it further we can use a new window as we did in my branch. It may mean dropping back to UIKit for the presentation but it's definitely animatable just like any UIView.

I'd avoided `fullscreenCover` because I thought it prevented us showing other `sheet` views over the top and caused issues with the `dismiss` environment object. From some quick tests on your PR that's not actually the case. We can even present another `fullscreenCover` over the top of this one.

Let's go with this approach because it's straightforward, unlike the new window one. Thanks!

**Comment by jaclync** [comment]
> Joe didn't love the slide-up transition... I think he wanted it to feel more like a mode switch. However, in the end we just went with the quickest thing, so I don't think the existing transition is thoroughly considered.
> 
> If we need to customise it further we can use a new window as we did in my branch. It may mean dropping back to UIKit for the presentation but it's definitely animatable just like any UIView.
> 
> I'd avoided `fullscreenCover` because I thought it prevented us showing other `sheet` views over the top and caused issues with the `dismiss` environment object. From some quick tests on your PR that's not actually the case. We can even present another `fullscreenCover` over the top of this one.
> 
> Let's go with this approach because it's straightforward, unlike the new window one. Thanks!

Good point on presenting sheets, I tested again with the support form, onboarding modal, and simple products info modal that navigates to the orders tab, and they worked as expected.

I tried the ZStack alternative (felt a bit hacky) in https://github.com/woocommerce/woocommerce-ios/commit/b29fc32043b34bf5a96696b7fb75a027d27b2636 that allows different animations, we can explore that in addition to the UIKit window presentation approach when we want to improve the transition experience.

**Pull request merged**